ox_bqpipeline/bqpipeline.py

```diff
@@ -350,72 +341,82 @@ def create_job_config(self, batch=False, dest=None, create=True,
         else:
             data_set = None
 
-        return bigquery.QueryJobConfig(priority=priority,
-                                       default_dataset=data_set,
-                                       destination=dest_tableref,
-                                       create_disposition=create_disp,
-                                       write_disposition=write_disp)
+        job_config_settings = {
+            'priority': priority,
+            'default_dataset': data_set,
+            'destination': dest_tableref,
+            'create_disposition': create_disp,
+            'write_disposition': write_disp,
+        }
+        if query_params:
+            job_config_settings.update({'query_parameters':
+                                        self.set_query_params(query_params)})
 
-    def validate_query_params(self, query_params, depth=0):
+        return bigquery.QueryJobConfig(**job_config_settings)
+
+    def validate_query_params(self, query_params):
         """Validate the named/positional query parameters."""
-        if not (isinstance(query_params, list) or
-                isinstance(query_params, dict)):
+        if not isinstance(query_params, (list, dict)):
             return False
-        for key in query_params:
-            if isinstance(query_params, list):
-                value = key
-                key = None
-            else:
-                if not (depth or isinstance(key, str)):
-                    return False
-                value = query_params[key]
-
+        if isinstance(query_params, list):
+            res = all([self.validate_parameter(value) for value in
```

Comment:
nit: more descriptive name 
s/res/all_params_valid/

Reply:
This would have to be the same variable that's returned at the end of the function. Instead, I have updated to return here itself.